api/notd/badge_manager.py

```diff
@@ -0,0 +1,49 @@
+import logging
+
+from core.queues.sqs_message_queue import SqsMessageQueue
+from core.store.retriever import StringFieldFilter
+from core.util import chain_util
+from core.util import list_util
+
+from notd.badge_processor import BadgeProcessor
+from notd.messages import RefreshAllUserBadgesMessageContent
+from notd.messages import RefreshUserBadgesForCollectionMessageContent
+from notd.model import GALLERY_COLLECTIONS
+from notd.store.retriever import Retriever
+from notd.store.saver import Saver
+from notd.store.schema import CollectionBadgeHoldersTable
+
+
+class BadgeManager:
+
+    def __init__(self, retriever: Retriever, saver: Saver, workQueue: SqsMessageQueue, badgeProcessor: BadgeProcessor) -> None:
+        self.retriever= retriever
+        self.saver= saver
+        self.workQueue= workQueue
+        self.badgeProcessor= badgeProcessor
+
+    async def refresh_all_users_badges_deferred(self):
+        await self.workQueue.send_message(message=RefreshAllUserBadgesMessageContent().to_message())
+
+    async def refresh_all_users_badges(self):
```

Comment:
this looks more like it should be refresh_user_badges_for_all_collections. What are you copying this from?